parachain/runtime/kintsugi/src/dex.rs

```diff
@@ -0,0 +1,247 @@
+use super::{
+    parameter_types, Balance, CurrencyId, DexGeneral, DexStable, DispatchResult, Farming, Get, Loans, OnRuntimeUpgrade,
+    PalletId, Rate, Ratio, Runtime, RuntimeEvent, RuntimeOrigin, StablePoolId, Timestamp, Tokens, Vec, Weight, KBTC,
+    KINT, KSM,
+};
+use sp_runtime::{traits::Zero, FixedPointNumber};
+
+#[cfg(feature = "try-runtime")]
+use frame_support::ensure;
+
+pub use dex_general::{AssetBalance, GenerateLpAssetId, PairInfo};
+pub use dex_stable::traits::{StablePoolLpCurrencyIdGenerate, ValidateCurrency};
+
+parameter_types! {
+    pub const DexGeneralPalletId: PalletId = PalletId(*b"dex/genr");
+    pub const DexStablePalletId: PalletId = PalletId(*b"dex/stbl");
+    pub const StringLimit: u32 = 50;
+}
+
+pub struct PairLpIdentity;
+impl GenerateLpAssetId<CurrencyId> for PairLpIdentity {
+    fn generate_lp_asset_id(asset_0: CurrencyId, asset_1: CurrencyId) -> Option<CurrencyId> {
+        CurrencyId::join_lp_token(asset_0, asset_1)
+    }
+}
+
+impl dex_general::Config for Runtime {
+    type RuntimeEvent = RuntimeEvent;
+    type MultiCurrency = Tokens;
+    type PalletId = DexGeneralPalletId;
+    type AssetId = CurrencyId;
+    type LpGenerate = PairLpIdentity;
+    type WeightInfo = ();
+}
+
+pub struct PoolLpGenerate;
+impl StablePoolLpCurrencyIdGenerate<CurrencyId, StablePoolId> for PoolLpGenerate {
+    fn generate_by_pool_id(pool_id: StablePoolId) -> CurrencyId {
+        CurrencyId::StableLpToken(pool_id)
+    }
+}
+
+pub struct StableAmmVerifyPoolAsset;
+impl ValidateCurrency<CurrencyId> for StableAmmVerifyPoolAsset {
+    fn validate_pooled_currency(_currencies: &[CurrencyId]) -> bool {
+        true
+    }
+
+    fn validate_pool_lp_currency(currency_id: CurrencyId) -> bool {
+        if Tokens::total_issuance(currency_id) > 0 {
+            return false;
+        }
+        true
+    }
+}
+
+impl dex_stable::Config for Runtime {
+    type RuntimeEvent = RuntimeEvent;
+    type CurrencyId = CurrencyId;
+    type MultiCurrency = Tokens;
+    type PoolId = StablePoolId;
+    type TimeProvider = Timestamp;
+    type EnsurePoolAsset = StableAmmVerifyPoolAsset;
+    type LpGenerate = PoolLpGenerate;
+    type PoolCurrencySymbolLimit = StringLimit;
+    type PalletId = DexStablePalletId;
+    type WeightInfo = ();
+}
+
+impl dex_swap_router::Config for Runtime {
+    type RuntimeEvent = RuntimeEvent;
+    type StablePoolId = StablePoolId;
+    type Balance = Balance;
+    type CurrencyId = CurrencyId;
+    type NormalAmm = DexGeneral;
+    type StableAMM = DexStable;
+    type WeightInfo = ();
+}
+
+pub struct DexSetup;
+
+fn create_pair_and_update_rewards(asset_0: CurrencyId, asset_1: CurrencyId, rewards: Balance) {
+    // TODO: should we bootstrap?
```

Comment:
Open point for @philippb90, should we bootstrap these markets instead of immediately initiating trading? Should there be bootstrap rewards associated with this as well?

Reply:
No, I don't see why.